### PR TITLE
Header icon to SVG

### PR DESCRIPTION
**Pull request by allcaps**
https://github.com/wagtail/wagtail/issues/6107
`Header icons <h1 class="icon icon-image">Images</h1>`

Do the tests still pass? YES
Does the code comply with the style guide? YES
For Python changes: Have you added tests to cover the new/fixed behaviour? N/A
For front-end changes: Did you test on all of Wagtail’s supported browsers?
- Google Chrome OSX Version 81
For new features: Has the documentation been updated accordingly? N/A

### CODE REVIEW

File: client/scss/components/_header.scss [hunk 1]
```diff
@@ -164,4 +164,11 @@ header {
     @include media-breakpoint-down(xs) {
         padding-left: $mobile-nav-indent;
     }
+
+    &-icon {
+        vertical-align: text-top;
+        margin-right: 0.5em;
+        width: 1.5em;
+        height: 1.5em;
```
Comment (zerolab) [inline client/scss/components/_header.scss:172]: This snippet seems to be recurring in the font icon to SVG PRs. I do wonder if we could have a sensible along the lines of:

```css
svg.icon {
    width: 1.5em;
    height: 1.5em;
    vertical-align: text-top;
}
```

then override where needed like here.

File: wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html [hunk 2]
```diff
@@ -14,7 +14,9 @@
 
 {% block content %}
     {% trans "Styleguide" as title_trans %}
-    {% include "wagtailadmin/shared/header.html" with title=title_trans %}
+    {% with icon='image' %}
+      {% include "wagtailadmin/shared/header.html" with title=title_trans %}
```
Comment (zerolab) [inline wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html:18]: `{% include "wagtailadmin/shared/header.html" with title=title_trans icon='image' %}` ?

Comment (allcaps) [reply wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html:18]: Yes, this is in the website frontend, not in the Wagtail admin. So the icon symbol needs to be added.

See the source of Wagtail admin page too. It contains all icons.

Comment (zerolab) [reply wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html:18]: Sorry, should have been explicit.. I meant it you could fold the `{% with %}` into the include. Not a biggie

File: client/scss/components/_header.scss (hunk 1 above)
Comment (zerolab) [reply client/scss/components/_header.scss:172]: having played with this a bit, may we provide a few modifiers something like:

```scss
.icon {
    &.initial {
        width: 1em;
        height: 1em;
        vertical-align: initial;
    }

    &.default {
        width: 1.5em;
        height: 1.5em;
        vertical-align: text-top;
    }
}
```

?

Comment (allcaps) [reply client/scss/components/_header.scss]: @zerolab Thanks for the feedback. I added a `svg-icon` mixin. With a default size.

File: wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html (hunk 2 above)
Comment (allcaps) [reply wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html]: Yes! Done.

File: wagtail/admin/templates/wagtailadmin/pages/create.html [hunk 3]
```diff
@@ -12,8 +12,10 @@
 
         <div class="row row-flush">
             <div class="left col9 header-title">
-                <h1 class="icon icon-doc-empty-inverse">
-                {% trans 'New' %} <span>{{ content_type.model_class.get_verbose_name }}</span></h1>
+                <h1>
+                  {% icon "doc-empty-inverse" %}
```
Comment (zerolab) [inline wagtail/admin/templates/wagtailadmin/pages/create.html:16]: missing `class_name="header-title-icon"`

other than that, works like a charm!

File: wagtail/contrib/settings/templates/wagtailsettings/edit.html [hunk 4]
```diff
@@ -7,7 +7,8 @@
         <div class="row">
             <div class="left">
                 <div class="col">
-                    <h1 class="icon icon-cogs">
+                    <h1>
+                        {% icon name="icon-cogs" class_name="header-title-icon" %}
```
[the remark this answers is not in the document]
Comment (allcaps) [reply wagtail/contrib/settings/templates/wagtailsettings/edit.html:11]: Done

**Comment by allcaps** [comment]
@zerolab and @Scotchester, the feedback is applied. I think this can be merged now.

**Comment by zerolab** [comment]
Thank you @allcaps,

merged (and release notes added in 18fdd90)

**Pull request closed without merging**